core/model/identifier.py

```diff
@@ -752,6 +752,24 @@ def add_link(
             resource.representation, is_new = get_one_or_create(
                 _db, Representation, url=resource.url, media_type=media_type
             )
+        elif (
+            media_type
+            and resource.representation
+            and resource.representation.media_type != media_type
+        ):
+            # Ensure we do not violate unique constraints
+            representation_exists = (
+                _db.query(Representation)
+                .filter(
+                    Representation.url == resource.url,
+                    Representation.media_type == media_type,
+                )
+                .count()
+            )
+            if not representation_exists:
+                # We have a representation that is not the same media type we previously knew of
+                resource.representation.media_type = media_type
+                resource.representation.url = resource.url
```

Comment:
This should be fine for now, but could be a problem longer term. For example, what would happen if OD moved to a content negotiation approach, using the same URL but with differing media types to determine the HTTP response for differing sample formats.

If there's not a good way to handle this now, it would be helpful to put a TODO or other note here to acknowledge that possibility.

Reply:
This depends on the replacement policy defined during metadata.apply()
https://github.com/ThePalaceProject/circulation/blob/main/core/metadata_layer.py#L1937
If it is set to replace, then its a complete replacement of all identifier links, so its a clean list every import.